### ISSUE TITLE
Error with eddy_cuda with option -repol

### ISSUE THREAD

**Issue by julfou81** [PennLINC/qsiprep#326]
I encounter this error message when running qsiprep v0.14.3 on a singularity container
```
EDDY:::  DoVolumeToVolumeRegistration: Unable to find volume with no outliers in shell 0 with b-value=300
```
The data set consists of 6b=0, b=300, 32 b=1000, 64 b =2000.

This issue is a specific `eddy` issue, reported elsewhere:
https://www.jiscmail.ac.uk/cgi-bin/wa-jisc.exe?A2=ind1912&L=FSL&D=0&P=107520

It looks like this can happen for such datasets with a few directions with small b.
A solution is to re-run the problematic runs by passing to eddy the hidden option: `--ol_ec=2`. I don't think this is possible through qsiprep?
Another method to deal with this problem in shown in *MRtrix3_connectome*:

```
            # If dwifslpreproc fails due to:
            # EDDY:::  DoVolumeToVolumeRegistration: Unable to find volume
            #          with no outliers in shell 0 with b-value=549.375
            # , want to progressively increase the outlier rejection threshold
            #   until this error no longer occurs.
            eddy_all_options = eddy_options + eddy_olnstd_option
            dwifslpreproc_eddy_option = \
                ' -eddy_options " ' \
                + ' '.join(eddy_all_options) + '"' if eddy_all_options else ''
```
https://github.com/BIDS-Apps/MRtrix3_connectome/blob/master/mrtrix3_connectome.py

In that case one can increase the default option `_ol_nstd=4` to 4.5 or 5.

I tried to do so by passing a new value within the eddy configuration file:

```
{
  "flm": "quadratic",
  "slm": "linear",
  "fep": true,
  "interp": "spline",
  "nvoxhp": 1000,
  "fudge_factor": 10,
  "dont_sep_offs_move": false,
  "dont_peas": true,
  "niter": 5,
  "method": "jac",
  "repol": true,
  "num_threads": 4,
  "is_shelled": true,
  "use_cuda": true,
  "cnr_maps": true,
  "residuals": true,
  "output_type": "NIFTI_GZ",
  "args": "","outlier_nstd": 5, "estimate_move_by_susceptibility": true, "mporder": 10, "multiband_factor":3
}
```
but I ended with the following error:

```
Node: qsiprep_wf.single_subject_testpilote6_wf.dwi_preproc_ses_03_wf.hmc_sdc_wf.eddy
Working directory: /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/hmc_sdc_wf/eddy

Node inputs:

args = 
cnr_maps = True
dont_peas = True
dont_sep_offs_move = False
environ = {'FSLOUTPUTTYPE': 'NIFTI_GZ', 'OMP_NUM_THREADS': '8'}
estimate_move_by_susceptibility = True
fep = True
field = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/hmc_sdc_wf/topup/fieldmap_HZ.nii.gz
field_mat = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/hmc_sdc_wf/topup_to_eddy_reg/topup_reg_image_flirt.mat
flm = quadratic
fudge_factor = 10.0
fwhm = <undefined>
in_acqp = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/hmc_sdc_wf/gather_inputs/eddy_acqp.txt
in_bval = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/pre_hmc_wf/rpe_concat/merge__merged.bval
in_bvec = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/pre_hmc_wf/rpe_concat/merge__merged.bvec
in_file = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/pre_hmc_wf/rpe_concat/merge__merged.nii.gz
in_index = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/hmc_sdc_wf/gather_inputs/eddy_index.txt
in_mask = /work/temp_data_PhantomPain/qsiprep_wf/single_subject_testpilote6_wf/dwi_preproc_ses_03_wf/hmc_sdc_wf/pre_eddy_b0_ref_wf/enhance_and_mask_b0/topup_imain_corrected_avg_mask.nii.gz
in_topup_fieldcoef = <undefined>
in_topup_movpar = <undefined>
initrand = <undefined>
interp = spline
is_shelled = True
json = <undefined>
mbs_ksp = <undefined>
mbs_lambda = <undefined>
mbs_niter = <undefined>
method = jac
mporder = 10
multiband_factor = 3
multiband_offset = <undefined>
niter = 5
num_threads = 8
nvoxhp = 1000
out_base = eddy_corrected
**outlier_nstd = 5**
outlier_nvox = <undefined>
outlier_pos = <undefined>
outlier_sqr = <undefined>
outlier_type = <undefined>
output_type = NIFTI_GZ
repol = True
residuals = True
session = <undefined>
slice2vol_interp = <undefined>
slice2vol_lambda = <undefined>
slice2vol_niter = <undefined>
slice_order = <undefined>
slm = linear
use_cuda = True

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 516, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 723, in _run_command
    cmd = self._interface.cmdline
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 732, in cmdline
    allargs = [self._cmd_prefix + self.cmd] + self._parse_inputs()
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 988, in _parse_inputs
    arg = self._format_arg(name, spec, value)
  File "/usr/local/miniconda/lib/python3.7/site-packages/qsiprep/interfaces/eddy.py", line 231, in _format_arg
    return super(ExtendedEddy, self)._format_arg(name, spec, value)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/fsl/epi.py", line 1006, in _format_arg
    return super(Eddy, self)._format_arg(name, spec, value)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 862, in _format_arg
    return argstr % value
TypeError: not all arguments converted during string formatting
```

The **new outlier_nstd** threshold was passed to eddy but there is still an error that I don't understand.

Please let me know what could be a workaround for this issue. I guess I could run without outlier replacement but that is not my goal here.

**Comment by mattcieslak** [comment]
how about setting `--b0-threshold 500` in your qsiprep call? You don't have enough samples in your b=300 shell to even fit a tensor, so you could just lump them in with the b=0 images.

**Comment by rwmair** [comment]
I encountered this same problem with scans acquired with the ABCD vector set that has a small number of b=500 volumes. I was able to get the ol_nstd value passed and everything ran successfully. I figured a number of additional eddy parameters including mporder and estimate_move_by_susceptibility can be set in the .json as their own entry without having to cram them all in the "args" line. Maybe that will help?  For what its worth, I tried with ol_ec=2 in the "args" line and that didn't solve the problem, hence I moved on to increasing ol_nstd.

Note this uses the default ol_type = slicewise; I couldn't get the ol_type parameter recognized itself in the .json file... if you want ol_type=gw, that probably has to go in the "args" line also.

`{
  "flm": "quadratic",
  "slm": "linear",
  "fep": false,
  "interp": "spline",
  "nvoxhp": 1000,
  "fudge_factor": 10,
  "dont_sep_offs_move": false,
  "dont_peas": false,
  "niter": 5,
  "method": "jac",
  "repol": true,
  "num_threads": 1,
  "is_shelled": true,
  "use_cuda": true,
  "cnr_maps": true,
  "residuals": true,
  "output_type": "NIFTI_GZ",
  "estimate_move_by_susceptibility": true,
  "mporder": 8,
  "slice_order": "/ncf/mri/01/users/rmair/multiband/211108_SRA_Diff_Human_08/slspec_ABCD_81sl.txt",
  "args": "--ol_nstd=5"
}
`
